### PR TITLE
feat(PDRIVE-806): replace all-namespaces pod check with scoped infra rule

### PR DESCRIPTION
**Pull request by sarad-rh**
## Ticket
https://redhat.atlassian.net/browse/PDRIVE-806

## Description
Replace `AllPodsReadyAndRunning` (queries all namespaces) with `InfraPodsReadyAndRunning` scoped to 22 critical OpenShift infrastructure namespaces. This avoids scale/memory issues and false positives from user workload pods.

## Changes
- Added `InfraPodsReadyAndRunning` rule with per-namespace queries (22 namespaces)
- Old (>24h) Failed Job/CronJob pods silently skipped (ephemeral by design)
- Old (>24h) Failed non-Job pods surfaced as WARNING (e.g. installer pods)
- Mixed scenario (new failures + old failed): FAILED with both sections separated
- `AllPodsReadyAndRunning` commented out in domain (not removed)

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated Kubernetes validation to focus on infrastructure namespaces only, replacing the prior “all pods” readiness logic.
  * Improved pod-health evaluation with UTC-based “historical” handling for failed infra pods, producing clearer **fail/warn/pass** outcomes.
* **Tests**
  * Added comprehensive unit coverage for the new infra-scoped validation, including readiness-by-container and pod-age edge cases.
* **Documentation**
  * Refreshed Confluence rule-page creation guidance, including updated template examples and MCP-based page creation workflow.

### CODE REVIEW

File: src/in_cluster_checks/rules/k8s/k8s_validations.py [hunk 1]
```diff
@@ -99,6 +100,154 @@ def _get_pods_lists(self):
...
+    """Verify pods in infrastructure namespaces are ready and running.
+
+    Scoped replacement for AllPodsReadyAndRunning that only checks critical
+    OpenShift infrastructure namespaces, avoiding scale/memory issues and
+    false positives from user workload pods.
+    """
+
+    objective_hosts = [Objectives.ORCHESTRATOR]
+    unique_name = "infra_pods_ready_and_running"
+    title = "Verify infrastructure pods are ready and running"
```
[the remark this answers is not in the document]
Comment (sarad-rh) [reply src/in_cluster_checks/rules/k8s/k8s_validations.py:112]: The default `supported_profiles = {"general"}` is correct for this rule — it applies to all OpenShift clusters (infrastructure namespaces exist on every cluster type). Confluence page will be created as part of a follow-up ticket.

File: src/in_cluster_checks/rules/k8s/k8s_validations.py [hunk 2]
```diff
@@ -99,6 +100,154 @@ def _get_pods_lists(self):
...
+
+                pod_info = {
+                    "namespace": namespace,
+                    "name": pod_name,
+                    "status": phase,
+                    "ready": ready_str,
+                }
+
+                if phase != "Running" or ready_containers != total_containers:
+                    not_running_pods.append(pod_info)
```
[the remark this answers is not in the document]
Comment (sarad-rh) [reply src/in_cluster_checks/rules/k8s/k8s_validations.py:220]: Fixed — added `total_containers == 0` to the not-running condition, with a test (`test_pod_with_zero_containers_not_running`).

File: src/in_cluster_checks/rules/k8s/k8s_validations.py [hunk 3]
```diff
@@ -99,6 +100,154 @@ def _get_pods_lists(self):
...
+
+    @staticmethod
+    def _is_old_pod(pod_data: dict, threshold_seconds: int) -> bool:
+        """Check if a pod is older than the given threshold.
+
+        Uses completionTime if available, otherwise falls back to creationTimestamp.
+        """
+        timestamp_str = pod_data.get("status", {}).get("completionTime") or pod_data.get("metadata", {}).get(
+            "creationTimestamp"
+        )
```
[the remark this answers is not in the document]
Comment (sarad-rh) [reply src/in_cluster_checks/rules/k8s/k8s_validations.py]: Fixed — `_is_old_pod` now iterates `containerStatuses[*].state.terminated.finishedAt` with `creationTimestamp` as fallback. Test helper updated accordingly (`completion_time` → `finished_at`).

File: src/in_cluster_checks/rules/k8s/k8s_validations.py [hunk 4]
```diff
@@ -99,6 +100,154 @@ def _get_pods_lists(self):
...
+        if not timestamp_str:
+            return False
+        try:
+            parsed_time = datetime.fromisoformat(timestamp_str.replace("Z", "+00:00"))
+            age_seconds = (datetime.now(timezone.utc) - parsed_time).total_seconds()
+            return age_seconds > threshold_seconds
+        except (ValueError, AttributeError):
+            return False
```
[the remark this answers is not in the document]
Comment (sarad-rh) [reply src/in_cluster_checks/rules/k8s/k8s_validations.py]: Fixed — the `except` block now raises `UnExpectedSystemOutput` with the raw timestamp string instead of silently returning `False`.

File: tests/rules/k8s/test_k8s_validations.py [hunk 5]
```diff
@@ -108,6 +109,215 @@ def test_no_pods_found(self, tested_object):
...
+
+    def test_only_queries_infra_namespaces(self, tested_object):
+        """Test that get_pods is only called for infrastructure namespaces."""
+        tested_object.oc_api.get_pods = Mock(return_value=[])
+        tested_object.run_rule()
+
+        called_namespaces = [call.kwargs.get("namespace") for call in tested_object.oc_api.get_pods.call_args_list]
+        assert set(called_namespaces) == set(InfraPodsReadyAndRunning.INFRA_NAMESPACES)
+
+
```
[the remark this answers is not in the document]
Comment (sarad-rh) [reply tests/rules/k8s/test_k8s_validations.py:328]: `RuleTestBase` is designed for node-level `Rule` classes that mock shell commands via `CmdOutput`. `InfraPodsReadyAndRunning` is an `OrchestratorRule` that calls `oc_api.get_pods()` returning Python objects — it doesn't fit the `RuleTestBase`/`CmdOutput` pattern. The existing test style matches other `OrchestratorRule` tests in this file (e.g., `TestNodesAreReady`, `TestAllDeploymentsAvailable`).

File: src/in_cluster_checks/rules/k8s/k8s_validations.py [hunk 6]
```diff
@@ -99,6 +100,158 @@ def _get_pods_lists(self):
...
+        """Check if a pod is older than the given threshold.
+
+        Uses the most recent container finishedAt if available, otherwise falls back to creationTimestamp.
+        """
+        timestamp_str = None
+        for cs in pod_data.get("status", {}).get("containerStatuses", []):
+            finished_at = cs.get("state", {}).get("terminated", {}).get("finishedAt")
+            if finished_at:
+                timestamp_str = finished_at
+        if not timestamp_str:
```
[the remark this answers is not in the document]
Comment (sarad-rh) [reply src/in_cluster_checks/rules/k8s/k8s_validations.py]: Fixed — now collects all `finishedAt` timestamps and uses `max()` to pick the most recent one.

File: src/in_cluster_checks/rules/k8s/k8s_validations.py [hunk 7]
```diff
@@ -99,6 +100,158 @@ def _get_pods_lists(self):
...
+            timestamp_str = pod_data.get("metadata", {}).get("creationTimestamp")
+        if not timestamp_str:
+            return False
```
[the remark this answers is not in the document]
Comment (sarad-rh) [reply src/in_cluster_checks/rules/k8s/k8s_validations.py:247]: Intentional design choice — `return False` means "not old" which treats the pod as a recent failure and reports it in the FAILED result. This is the safe/conservative approach: alert on a pod we can't age rather than raising `UnExpectedSystemOutput` which would SKIP the entire rule and hide all other pod issues.

**Review by sprizend-rh — APPROVED** [review]

File: .claude/rules/confluence-guidelines.md [hunk 8]
```diff
@@ -21,8 +21,32 @@ Example location: `src/in_cluster_checks/rules/network/ovs_validations.py`
...
+```
+mcp__plugin_atlassian_atlassian__createConfluencePage(
+  cloudId: "redhat.atlassian.net",
+  spaceId: "377096307",
+  parentId: "<domain-parent-page-id>",
+  title: "<rule-title>",
+  contentFormat: "html",
+  body: "<page-content>"
+)
+```
```
[the remark this answers is not in the document]
Comment (sarad-rh) [reply .claude/rules/confluence-guidelines.md:49]: Already fixed — the fenced block on line 40 already has the `python` language identifier (see current file content). This comment appears to be from a stale review.

File: .claude/skills/implement-rule-from-jira/SKILL.md [hunk 9]
```diff
@@ -144,15 +144,26 @@ Ensure all tests pass before proceeding.
...
+**Page title**: Use the rule's `title` field exactly (e.g., "Verify infrastructure pods are ready and running").
+
+**Domain parent page IDs** (space ID: `377096307`):
+- DPF: `418450735`
+- Hardware: `418418340`
+- K8s: `418516155`
+- Linux: `418482835`
+- Network: `418482854`
+- Resources: `418450807`
+- Security: `418516403`
```
[the remark this answers is not in the document]
Comment (sarad-rh) [reply .claude/skills/implement-rule-from-jira/SKILL.md:164]: Storage and Etcd domains don't have dedicated Confluence parent pages yet — they'll be created when rules for those domains need documentation. The mapping intentionally only lists domains that currently exist in Confluence. The same list is used in `.claude/rules/confluence-guidelines.md`.

**Comment by sarad-rh** [comment]
@coderabbitai review
